src/realm/parser/CMakeLists.txt

```diff
@@ -56,13 +59,25 @@ set(REALM_PARSER_EXTRAS
     query_flex.ll
 )
 
+# Add custom warning flags for generated parser files if the flag is supported
+set(PARSER_COMPILE_EXTRA -Wno-unused-but-set-variable)
```

Comment:
that is too verbose. How about the function with ARGN for one option for multiple files which also checks for compiler flag underneath?